### ISSUE TITLE
ArrayInterface broken with Vitis 2020.1

### ISSUE THREAD

**Issue by definelicht** [spcl/dace#536]
After "upgrading" to Vitis 2020.1/2020.2, we get this error when writing to an output pointer extracted from an `ArrayInterface` object:

```
ERROR: [HLS 214-134] in function 'module_write_C_0(float const*, float*, dace::FIFO<float, 1u, 1u>*)': Pointer to pointer is not supported (/home/definelj/dev/dace/dace/codegen/../runtime/include/dace/xilinx/array_interface.h:15:9)
ERROR: [HLS 214-134] in function 'module_write_C_0(float const*, float*, dace::FIFO<float, 1u, 1u>*)': Pointer to pointer is not supported (/home/definelj/dev/dace/dace/codegen/../runtime/include/dace/xilinx/array_interface.h:15:26)
ERROR: [HLS 214-134] in function 'module_write_C_0(float const*, float*, dace::FIFO<float, 1u, 1u>*)': Pointer to pointer is not supported (/home/definelj/dev/dace/dace/codegen/../runtime/include/dace/xilinx/array_interface.h:41:9)
ERROR: [HLS 214-134] in function 'module_write_C_0(float const*, float*, dace::FIFO<float, 1u, 1u>*)': Pointer to pointer is not supported (/home/definelj/dev/dace/dace/codegen/../runtime/include/dace/xilinx/array_interface.h:47:53)
ERROR: [HLS 214-135] Syn check fail!
```

It seems it might be time to get rid of `ArrayInterface` once and for all.

**Comment by tbennun** [comment]
Reproduced and agreed.

**Comment by definelicht** [comment]
Fixed with #657 and #660.
